File: tests/fixtures/swift/complex.swift
import Foundation
import UIKit

class Calculator {
    private var value: Int = 0

    func add(a: Int, b: Int) -> Int {
        value = a + b
        return value
    }

    func printMessage(_ message: String) {
        print(message)
    }
}

struct Point {
    let x: Double
    let y: Double

    func distanceTo(_ other: Point) -> Double {
        let dx = x - other.x
        let dy = y - other.y
        return sqrt(dx * dx + dy * dy)
    }
}

protocol IProcessor {
    func process(data: String)
}

enum Color {
    case red
    case green
    case blue
}

typealias StringMap = Dictionary<String, String>

extension String {
    func trimmed() -> String {
        return self.trimmingCharacters(in: .whitespaces)
    }

    var isNullOrEmpty: Bool {
        return isEmpty
    }
}

func main() {
    let calc = Calculator()
    let result = calc.add(a: 5, b: 10)
    print(result)
}

File: tests/fixtures/dart/complex.dart
import 'dart:math';
import 'dart:async';
import 'package:flutter/material.dart';

class Calculator {
  int _value = 0;

  Calculator();

  int add(int a, int b) {
    _value = a + b;
    return _value;
  }

  void print(String message) {
    print(message);
  }
}

class Point {
  double x;
  double y;

  Point(this.x, this.y);

  double distanceTo(Point other) {
    return sqrt(pow(x - other.x, 2) + pow(y - other.y, 2));
  }
}

abstract class IProcessor {
  void process(String data);
}

enum Color {
  red,
  green,
  blue
}

mixin Printable {
  void printMessage();
}

extension StringExtensions on String {
  String trimmed() {
    return trim();
  }

  bool get isNullOrEmpty => isEmpty;
}

void main() {
  var calc = Calculator();
  int result = calc.add(5, 10);
  print(result);
}
